In [17]:
import numpy as np

In [18]:
a100_power_consumption = 250 # Watt
# https://bil24.no/tesla-model-y-long-range-firehjulsdrift-nordmenns-helt-klare-favoritt
tesla_kwh_per_km = np.mean([15.1, 20.1, 22.8]) / 100
print(tesla_kwh_per_km)

0.19333333333333333


Training times

In [19]:


yolo_training_time_v1 = 769.24 # seconds
yolo_traning_time_iphone = 1500 # seconds
rfdetr_training_time_v1 = None
rfdetr_training_time_iphone = None






Energy calculations

In [20]:
def joule_to_kwh(joule):
    return joule / 3_600_000

energy_yolo_v1 = joule_to_kwh(a100_power_consumption * yolo_training_time_v1)
energy_yolo_iphone = joule_to_kwh(a100_power_consumption * yolo_traning_time_iphone)

print(energy_yolo_v1, energy_yolo_iphone)




0.05341944444444444 0.10416666666666667


km calculations

In [21]:
km_tesla_yolo_v1 = energy_yolo_v1 / tesla_kwh_per_km
km_tesla_yolo_iphone = energy_yolo_iphone / tesla_kwh_per_km

print(km_tesla_yolo_v1, km_tesla_yolo_iphone)

0.27630747126436783 0.5387931034482759


You can drive 0.27 km with the amount of energy used traning the YOLO26s model on the v1 dataset with a A100 NVIDIA GPU. 

You can drive 0.54 km with the amount of energy used traning the YOLO26s model on the iphone dataset with a A100 NVIDIA GPU. 